extensions/frontend-tiles/src/graphics-provider/tileset-creators/IModelTileset.ts

```diff
@@ -0,0 +1,104 @@
+ * @see [[FrontendTilesOptions.computeSpatialTilesetBaseUrl]].
+ * @alpha
+ */
+export type ComputeSpatialTilesetBaseUrl = (iModel: IModelConnection) => Promise<URL | undefined>;
+
+/** Arguments supplied to [[CreateIModelTilesetAs3DTiles]]
+ * @alpha
+ */
+export type CreateIModelTilesetAs3DTilesArgs = GetIModelTilesetUrlFromConnectionArgs;
+
+/** Options supplied to [[createIModelTileset]].
+ * @alpha
+ */
+export interface CreateIModelTilesetArgs {
+  /** Provide the base URL for the pre-published tileset for a given iModel.
+   * If omitted, [[obtainMeshExportTilesetUrl]] will be invoked with default arguments, using the access token provided by [[IModelApp]].
+   */
+  computeSpatialTilesetBaseUrl?: ComputeSpatialTilesetBaseUrl;
+  /** The maximum number of levels in the tile tree to skip loading if they do not provide the desired level of detail for the current view.
+   * Default: 4.
+   * Reducing this value will load more intermediate tiles, which causes more gradual refinement: low-resolution tiles will display quickly, followed more gradually by
+   * successively higher-resolution ones.
+   * Increasing the value jumps more directly to tiles of the exact level of detail desired, which may load more, smaller tiles up-front, leaving some areas of the view
+   * vacant for longer; and when zooming out some newly-exposed areas of the view may remain vacant for longer because no lower-resolution tiles are initially available to
+   * fill them. However, tiles close to the viewer (and therefore likely of most interest to them) will refine to an appropriate level of detail more quickly.
+   */
+  maxLevelsToSkip?: number;
+  /** Specifies whether to permit the user to enable visible edges or wireframe mode for batched tiles.
+   * The currently-deployed mesh export service does not produce edges, so this currently defaults to `false` to avoid user confusion.
+   * Set it to `true` if you are loading tiles created with a version of the exporter that does produce edges.
+   * ###TODO delete this option once we deploy an edge-producing version of the exporter to production.
+   * @internal
+   */
+  enableEdges?: boolean;
+  /** Specifies whether to enable a CDN (content delivery network) to access tiles faster.
+   * This option is only used if computeSpatialTilesetBaseUrl is not defined.
+   * @alpha
+   */
+  enableCDN?: boolean;
+  /** Specifies whether to enable an IndexedDB database for use as a local cache.
+  * Requested tiles will then first be search for in the database, and if not found, fetched as normal.
+  * @internal
+  */
+  useIndexedDBCache?: boolean;
+}
+
+/** Global configuration initialized by [[initializeFrontendTiles]].
+ * @internal
+ */
+export const frontendTilesOptions = {
+  maxLevelsToSkip: 4,
+  enableEdges: false,
+  useIndexedDBCache: false,
+};
+
+/**
+ * Initializes the IModelTiles with the provided options.
+ * @alpha
+ */
+export function createIModelTileset(options: CreateIModelTilesetArgs): void {
```

Comment:
This does not create an iModel tileset. Who is expected to call this function?

Reply:
It would be called for example in `initializeFrontendTiles` or wherever that function is called. It does the same thing as
`initializeFrontendTiles`, essentially replacing/moving it. I agree that the name is misleading however. Our thought was to introduce this and similar functions for demo purposes or as a "first draft".

Reply:
I thought this PR was about enabling EVO. They don't need this, do they? And everyone else will continue to use the existing APIs like `initializeFrontendTiles`, right?

Reply:
That's correct, this was initially part of a proposed refactor from a previous PR which attempted to set up these sorts of functions for future use. There's no immediate use for it, but I believe it is indicative of the way @danieliborra wants this area of the code to look in the future. This is an example of @eringram's comment about the coupling of this overall refactoring and the seequent demo code. If all this PR is attempting to accomplish is to set up the seequent demo, then this is not necessary.